rustls/tests/api.rs

```diff
@@ -515,13 +519,33 @@ fn resumption_combinations() {
 
             assert_eq!(client.handshake_kind(), Some(HandshakeKind::Full));
             assert_eq!(server.handshake_kind(), Some(HandshakeKind::Full));
+            assert_eq!(
+                client
+                    .negotiated_key_exchange_group()
+                    .unwrap()
+                    .name(),
+                expected_kx
+            );
+            assert_eq!(
+                server
+                    .negotiated_key_exchange_group()
+                    .unwrap()
+                    .name(),
+                expected_kx
+            );
 
             let (mut client, mut server) =
                 make_pair_for_configs(client_config.clone(), server_config.clone());
             do_handshake(&mut client, &mut server);
 
             assert_eq!(client.handshake_kind(), Some(HandshakeKind::Resumed));
             assert_eq!(server.handshake_kind(), Some(HandshakeKind::Resumed));
+            assert!(client
+                .negotiated_key_exchange_group()
+                .is_none());
+            assert!(server
+                .negotiated_key_exchange_group()
```

Comment:
Interesting, I was expecting to see a `if version == &TLS12` somewhere around here.

Reply:
Aha yes, I think I should be updating `cx.common.kx_group` in `handle_server_hello` lower than where I have it. I'll push a fix.

Reply:
Fixed.